app/gui2/src/components/ComponentBrowser.vue

```diff
@@ -27,18 +28,38 @@ const props = defineProps<{
   navigator: ReturnType<typeof useNavigator>
   initialContent: string
   initialCaretPosition: ContentRange
+  sourceNode: Opt<ExprId>
 }>()
 
 const emit = defineEmits<{
   finished: [selectedExpression: string]
 }>()
 
+const initialContent = computed(() => {
+  if (props.sourceNode == null) return props.initialContent
+  const sourceNode = props.sourceNode
+  const sourceNodeName = useGraphStore().getNodeBinding(sourceNode)
+  const sourceNodeNameWithDot = sourceNodeName ? sourceNodeName + '.' : ''
+  return sourceNodeNameWithDot + props.initialContent
+})
```

Comment:
I think this and `initialCaret` does not need to be `computed`, functions should be enough: we want to get this information only at specific event of opening component browser.

Also, we do not actually expect to have both source node _and_ initial content (you create a node with source node known or edit an existing node without any source).